sdks/python/apache_beam/utils/timestamp.py

```diff
@@ -143,11 +148,35 @@ def __int__(self):
     # Note that the returned value may have lost precision.
     return self.micros // 1000000
 
-  def __cmp__(self, other):
-    # Allow comparisons between Duration and Timestamp values.
+  def __eq__(self, other):
```

Comment:
How about we keep `__eq__` and `__lt__`, remove the rest and decorate the class with `@functools.total_ordering`?